src/read.jl

```diff
@@ -39,33 +41,27 @@ to `identity`.
 edf_unknown(f, field::AbstractString) = field == "X" ? missing : f(field)
 edf_unknown(field::AbstractString) = edf_unknown(identity, field)
 
-"""
-    set!(io::IO, [f=strip,] sigs::Vector{Signal}, name::Symbol, sz::Integer)
+#####
+##### Reading EDF Files
+#####
 
-For each signal in `sigs`, set the field given by `name` to the result of `f` applied
-to a chunk of size `sz` read from `io` as a `String`.
-"""
-function set!(io::IO, f, sigs::Vector{Signal}, name::Symbol, sz::Integer)
-    n = length(sigs)
-    T = fieldtype(Signal, name)
-    @inbounds for i = 1:n
-        x = f(String(Base.read(io, sz)))
-        setfield!(sigs[i], name, convert(T, x))
+
+function read_header(io::IO)
```

Comment:
should this be renamed `read_file_and_signal_headers` or something? Would prefer that if it's more accurate/less ambiguous now that there are multiple header types floating around

Reply:
might also rename the signal headers and annotation headers during the mmap process to Signal and AnnotationListHeader, then do `read_signal` and `read_samples` a la Onda.

Reply:
Fixing this for now -- this may change in the mmap PR as we modify the design to be more Onda-like. https://github.com/beacon-biosignals/EDF.jl/pull/23/commits/0ea7f886aa5b5ef2109f2d47cd1253f638e515a4